### PR TITLE
REST cleanups

### PR DESCRIPTION
**Pull request by arnetheduck**
* reject out-of-range committee requests
* print all hex values as lower-case
* allow requesting state information by head state root
* turn `DomainType` into array (follow spec)
* `uint_to_bytesXX` -> `uint_to_bytes` (follow spec)
* fix wrong dependent root in `/eth/v1/validator/duties/proposer/`
* update documentation - `--subscribe-all-subnets` is no longer needed
when using the REST interface with validator clients (as of https://github.com/status-im/nimbus-eth2/pull/2949 which is included in 1.5.5)

### CODE REVIEW

File: beacon_chain/rpc/rest_validator_api.nim [hunk 1]
```diff
@@ -504,8 +511,6 @@ proc installValidatorApiHandlers*(router: var RestRouter, node: BeaconNode) =
   router.api(MethodPost,
              "/eth/v1/validator/beacon_committee_subscriptions") do (
     contentBody: Option[ContentBody]) -> RestApiResponse:
-    # TODO (cheatfate): This call could not be finished because more complex
-    # peer manager implementation needed.
```
Comment (tersec) [inline beacon_chain/rpc/rest_validator_api.nim:508]: Was that TODO done?

File: docs/the_nimbus_book/src/rest-api.md [hunk 2]
```diff
@@ -1,31 +1,29 @@
 # REST API
 
-Nimbus supports the [common REST API](https://ethereum.github.io/beacon-APIs/) for runtime communication. To enable it, use the `--rest` option when starting the beacon node, then access the API from http://localhost:5052/. The port and listening address can be further configured through the options `--rest-port` and `--rest-address`.
+Nimbus exposes a high-performance implementation of the [Beacon Node API](https://ethereum.github.io/beacon-APIs/).
 
-The API is a REST interface, accessed via HTTP. The API should not, unless protected by additional security layers, be exposed to the public Internet as the API includes multiple endpoints which could open your node to denial-of-service (DoS) attacks through endpoints triggering heavy processing.
+The API is a `REST` interface accessed via `HTTP`. The API should not, unless protected by additional security layers, be exposed to the public Internet as the API includes multiple endpoints which could open your node to denial-of-service (DoS) attacks through endpoints triggering heavy processing.
```
Comment (unixpi) [inline docs/the_nimbus_book/src/rest-api.md]:
```suggestion
The API is a `REST` interface accessed via `HTTP`. The API should not, unless protected by additional security layers, be exposed to the public Internet as the API includes multiple endpoints which could open your node to denial-of-service (DoS) attacks.
```

File: docs/the_nimbus_book/src/rest-api.md [hunk 3]
```diff
@@ -1,31 +1,29 @@
...
 
-The beacon node (BN) maintains the state of the beacon chain by communicating with other beacon nodes in the Ethereum network. Conceptually, it does not maintain keypairs that participate with the beacon chain.
+The API can be used with any conforming consumer, including alternative validator client implementations, explorers and tooling.
 
-The validator client (VC) is a conceptually separate entity which utilizes private keys to perform validator related tasks, called "duties", on the beacon chain. These duties include the production of beacon blocks and signing of attestations.
+## Configuration
 
-> Please note that using a validator client with a Nimbus beacon node requires the node to be launched with the `--subscribe-all-subnets` option. This limitation will be removed in Nimbus 1.6.
+By default, the REST interace is disabled. To enable it, use the `--rest` option when starting the beacon node, then access the API from http://localhost:5052/.
```
Comment (unixpi) [inline docs/the_nimbus_book/src/rest-api.md]:
```suggestion
By default, the REST interface is disabled. To enable it, start the beacon node with the `--rest` option, then access the API from http://localhost:5052/.

> **Warning:** If you are using a validator client with a Nimbus beacon node, and running a Nimbus version prior to `v1.5.5`,  then you will need to launch the node with the `--subscribe-all-subnets` option enabled.
```

**Review by unixpi — COMMENTED** [review]
docs review

**Pull request merged**